src/celeritas/ext/detail/GeantGeometryImporter.hh

```diff
@@ -0,0 +1,145 @@
+    VPlacedVolume const* world_;
+
+    // one G4 physical volume can correspond to multiple vecgeom placed volumes
+    // (in case of replicas)
+    BidirectionalTypeMap<std::vector<VPlacedVolume const*> const*,
+                         G4VPhysicalVolume const*>
+        placed_volume_map_;
+
+    BidirectionalTypeMap<VUnplacedVolume const*, G4VSolid const*>
+        unplaced_volume_map_;
+
+    BidirectionalTypeMap<LogicalVolume const*, G4LogicalVolume const*>
+        logical_volume_map_;
+
+    // fast O(1) lookup to get VecGeom or G4 placed volume based on index
+    // FastG4VecGeomLookup fFastG4VGLookup;
+    std::vector<Transformation3D const*> replica_transformations_;
+
+    int verbose_;
+
+  public:
+    void set_verbose(int const verbose) { verbose_ = verbose; }
+    int get_verbose_level() const { return verbose_; }
+
+    /// Get placed volume that corresponds to a G4VPhysicalVolume
+    std::vector<VPlacedVolume const*> const*
+    get_placed_volume(G4VPhysicalVolume const* n) const
+    {
+        if (n == nullptr)
+            return nullptr;
+        // return (GeoManager::Instance().Convert(placed_volume_map_[n]));
+        assert(false);
+        return nullptr;
+    }
+
+    /**
+     * Queries the G4 geometry for the top volume and recursively
+     * imports and converts to VecGeom geometry.
+     */
+    void convert_G4_geometry(G4VPhysicalVolume const*);
+
+  private:
+    /**
+     * @brief Deletes all VecGeom geometry generated by this class.
+     */
+    void clear_vecgeom();
+
+    /** @brief Converts a physical volume into a VecGeom placed volume.
+     * Its transformation matrix and its logical volume are also converted,
+     * making the conversion process recursive, comprising the whole geometry
+     * starting from the top volume.
+     * Will take care not to convert anything twice by checking the
+     * bidirectional map between Geant4 and VecGeom geometry.
+     */
+    std::vector<VPlacedVolume const*> const* convert(G4VPhysicalVolume const*);
+
+    /**
+     * @brief Special treatment needed for replicated volumes.
+     */
+    void extract_replicated_transformations(
+        G4PVReplica const&, std::vector<Transformation3D const*>&) const;
+
+    /**
+     * @brief Converts G4 solids into VecGeom unplaced volumes
+     */
+    VUnplacedVolume* convert(G4VSolid const*);
+
+    /**
+     * @brief Converts logical volumes from Geant4 into VecGeom.
+     * All daughters' physical volumes will be recursively converted.
+     */
+    LogicalVolume* convert(G4LogicalVolume const*);
+
+    /**
+     * @brief Converts transformation matrices
+     */
+    Transformation3D* convert(G4ThreeVector const&, G4RotationMatrix const*);
+
+  public:
+    // constructor
+    GeantGeometryImporter()
+        : world_(nullptr)
+        , placed_volume_map_()
+        , unplaced_volume_map_()
+        , logical_volume_map_()
+        , replica_transformations_()
+        , verbose_(0)
+    {
+    }
```

Comment:
These should all probably use C++ default initializers and this should be `= default`.